framework/src/org/checkerframework/framework/type/TypeHierarchy.java

```diff
@@ -8,13 +10,58 @@
  */
 public interface TypeHierarchy {
 
+
     /**
-     * @return true if subtype {@literal <:} supertype for all qualifier hierarchies in the type system
+     * Returns true if {@code subtype} is a subtype of or convertible to {@code supertype}
+     * for all hierarchies present.  If the underlying Java type of {@code subtype} must be a subtype of
+     * or convertible to the underlying Java type of {@code supertype}, then the behavior of this method is undefined.
```

Comment:
I don't understand when the behavior is undefined. Usually the underlying type of `subtype` will be a subtype of the underlying type of `supertype`. Why would this be undefined?